apis/server/router.go

```diff
@@ -106,6 +106,35 @@ func profilerSetup(mainRouter *mux.Router) {
 
 type handler func(context.Context, http.ResponseWriter, *http.Request) error
 
+// withCancelHandler will use context to cancel the handler. Otherwise, if the
+// the connection has been cut by the client or firewall, the server handler
+// will hang and cause goroutine leak.
+func withCancelHandler(h handler) handler {
+	return func(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+		if notifier, ok := rw.(http.CloseNotifier); ok {
```

Comment:
How about using `return fast` to make less ident?

```
if notifier, ok := rw.(http.CloseNotifier); !ok {
    return h(ctx, rw, req)
}
var cancel context.CancelFunc
ctx, cancel = context.WithCancel(ctx)
......
return h(ctx, rw, req)
```

Reply:
Agree. Will update it.